raiden/tests/utils/smoketest.py

```diff
@@ -249,7 +248,7 @@ def setup_testchain(
 
     eth_rpc_endpoint = URI(f"http://127.0.0.1:{rpc_port}")
     web3 = Web3(HTTPProvider(endpoint_uri=eth_rpc_endpoint))
-    web3.middleware_onion.inject(geth_poa_middleware, layer=0)
+    web3.middleware_onion.inject(make_sane_poa_middleware, layer=0)
```

Comment:
Don't you need to call `make_sane_poa_middleware` here?

Reply:
No, web3 expects a callable that returns the actual middleware callable.

If the generating wrapper would need to take arguments it would need another layer (similar to how decorators with and without arguments) and then would need to be called.

Reply:
👍